spec-main/api-protocol-spec.ts

```diff
@@ -223,7 +223,7 @@ describe('protocol module', () => {
       expect(r.headers).to.include('x-great-header: sogreat')
     })
 
-    it('throws an error when custom headers are invalid', (done) => {
+    it.skip('throws an error when custom headers are invalid', (done) => {
```

Comment:
We are now skipping invalid headers https://github.com/electron/electron/blob/master/shell/browser/net/atom_url_loader_factory.cc#L112-L122, this spec was added recently https://github.com/electron/electron/pull/18854. I am fine with restoring the behavior but I feel that rather than throwing exception , sanitizing should be more of ignoring invalid values.

@zcbenz wdyt ?

Reply:
For protocol APIs, we usually send network errors when there is invalid input, we didn't throw exceptions under any case.